include/gatekeeper_net.h

```diff
@@ -443,11 +443,67 @@ int convert_ip_to_str(const struct ipaddr *ip_addr, char *res, int n);
 int ethertype_filter_add(struct gatekeeper_if *iface, uint16_t ether_type,
 	uint16_t queue_id);
 
-int ntuple_filter_add(struct gatekeeper_if *iface, uint32_t dst_ip,
+/*
+ * Add a filter for IPv4 packets based on the destination IP address,
+ * source and destination ports, and protocol.
+ *
+ * @dst_ip: destination IP address to match against
+ * @src_port: L4 source port to match against
+ * @src_port_mask: mask for @src_port; set to 0xFFFF to match
+ *     against @src_port, or set to 0 to not match against @src_port
+ * @dst_port: L4 destination port to match against
+ * @dst_port_mask: mask for @dst_port; set to 0xFFFF to match
+ *     against @dst_port, or set to 0 to not match against @dst_port
+ * @proto: next header protocol to match against
+ *
+ * Filters will be installed either as ntuple filters (if supported
+ * by the NIC) or ACLs (as a software backup).
+ *
+ * @queue_id: the RX queue ID to which matching packets are
+ *     steered (if ntuple filters are supported)
+ * @cb_f: the function that is invoked on matching packets
+ *     (if the ACL is being used)
+ * @ext_cb_f: the function that is invoked on potential
+ *     matching packets that checks for variable-length/extension
+ *     headers (if the ACL is being used)
+ */
+int ipv4_ntuple_or_acl_add(struct gatekeeper_if *iface, uint32_t dst_ip,
 	uint16_t src_port, uint16_t src_port_mask,
 	uint16_t dst_port, uint16_t dst_port_mask,
```

Comment:
It seems that all calls of `ipv4_ntuple_or_acl_add()` assume that parameters `src_port`, `src_port_mask`, `dst_port`, and `dst_port_mask` are big endian. That's consistent with the old `ntuple_filter_add()`. However, ACLs work with host endian.

1. Notice endianness in the description of the function;
2. Update the type of the parameters that are affected by endianness to `rte_be16_t` to document the code;
3. Check that all calls respect endianness;
4. Review the function to make sure that it's respecting endianness when calling other functions.

This comment also applied to `ipv6_ntuple_or_acl_add()`.